### PR TITLE
Fix dead link in nodes-machines-and-hosts.md

### PR DESCRIPTION
**Pull request by geekrypter**
What is the change?
This change fixes a dead link in nodes-machines-and-hosts.md, and there is a minor text change.

Why the change?
Fixes: #441

### CODE REVIEW

**Review by tuminoid — COMMENTED** [review]
/ok-to-test
/lgtm

File: design/nodes-machines-and-hosts.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ such as draining the node for maintenance. The definition of Node is
 owned by the Kubernetes core, and offers few opportunities to attach
 custom data.
 
-[Machine](https://kubernetes-sigs.github.io/cluster-api/common_code/machine_controller.html)
+[Machine](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine)
```
Comment (zhouhao3) [inline design/nodes-machines-and-hosts.md:60]: It should be https://cluster-api.sigs.k8s.io/developer/core/controllers/machine

Comment (geekrypter) [reply design/nodes-machines-and-hosts.md]: Made this change in the latest commit

**Comment by tuminoid** [comment]
Squash the commits, then its good to go.

**Comment by geekrypter** [comment]
> Squash the commits, then its good to go.

Squashed the commits.

**Review by tuminoid — COMMENTED** [review]
LGTM

/cc @kashifest @Rozzii @elfosardo

**Review by Rozzii — COMMENTED** [review]
/approve

**Pull request merged**